### ISSUE TITLE
🧹 [Maintenance]: Update docs to use a consistent indentation style

### ISSUE THREAD

**Issue by grivera64** [gofiber/fiber#2955]
### Maintenance Task Description

In the docs, there are some inconsistencies in indentation styles (e.g. `\t`, 4 spaces, 2 spaces, etc...).

For example:
- [`rewrite`](https://github.com/gofiber/fiber/blob/main/docs/api/middleware/rewrite.md) middleware uses 2-space tabs
- [`session`](https://github.com/gofiber/fiber/blob/main/docs/api/middleware/session.md) uses 4-space tabs
- [`idempotency`](https://github.com/gofiber/fiber/blob/main/docs/api/middleware/idempotency.md) middleware uses `\t` tabs

For readability, this issue proposes to update the docs to use a consistent indentation style to avoid any potential formatting issues.

### Impact on the Project

While the tabs versus `n`-spaces debate is very controversial and will most likely never be agreed upon, there are problems with inconsistent indentation within a project.

For example, using inconsistent spacing in a single file cause code snippets to render in https://gofiber.io in an unaligned manner:
- [`context`](https://github.com/gofiber/fiber/blob/main/docs/api/ctx.md?plain=1#L324) uses both 2-, 4-, and 8-space tabs within the same file.
- [`compress`](https://github.com/gofiber/fiber/blob/main/docs/api/middleware/compress.md?plain=1#L42) middleware uses both 2- and 4-space tabs within the same file.

There are also smaller issues that have been made due to this issue in the past. For example, #2866 fixes `key_auth.md` that used 8-space indentation alongside 4-space indentation.

If GoFiber was to adopt a tab spacing standard, we can avoid most tab spacing inconsistencies issues within the docs in the future.

A potential risk would be that this may require a lot of refactoring work. This could be addressed in the long-term over several smaller issues, but this is largely a readability issue that is mostly prominent when two indentation styles are used in the same file. As long as individual files use a consistent tab style, this becomes less of an issue.

### Checklist:

- [X] I have confirmed that this maintenance task is currently not being addressed.
- [X] I understand that this task will be evaluated by the maintainers and prioritized accordingly.
- [X] I am available to provide further information if needed.

**Comment by ReneWerner87** [comment]
> I propose introducing the following actions in a new workflow:
> 
> * https://github.com/marketplace/actions/markdownlint-cli2-action
> * https://github.com/gaurav-nelson/github-action-markdown-link-check

good idea

**Comment by gaby** [comment]
I will give this a try tonight.

**Comment by grivera64** [comment]
> > I propose introducing the following actions in a new workflow:
> > 
> > * https://github.com/marketplace/actions/markdownlint-cli2-action
> > * https://github.com/gaurav-nelson/github-action-markdown-link-check
> 
> good idea

I agree, this sounds like a good way to automate markdown format review (especially in cases of 4-spaces vs `\t` renders the same in VS Code but not on the gofiber.io docs), as well as check other markdown-related formatting inconsistencies.

This will only bring two more things we need to address to add these to GitHub Actions:
- What indentation format will we enforce? We can customize this in the config, or use default settings and enforce it from now on.
- It also seems that we will need to also manually update the docs to comply with the linter rules, based on their GitHub repository (Please correct me if this assumption is wrong).

What do you all think about the above?
